src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector256.cs

```diff
@@ -49,72 +49,58 @@ public static bool IsHardwareAccelerated
         }
 
         /// <summary>Computes the absolute value of each element in a vector.</summary>
-        /// <param name="vector">The vector that will have its absolute value computed.</param>
         /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+        /// <param name="vector">The vector that will have its absolute value computed.</param>
         /// <returns>A vector whose elements are the absolute value of the elements in <paramref name="vector" />.</returns>
+        /// <exception cref="NotSupportedException">The type of <paramref name="vector" /> (<typeparamref name="T" />) is not supported.</exception>
         [Intrinsic]
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector256<T> Abs<T>(Vector256<T> vector)
             where T : struct
         {
-            if (typeof(T) == typeof(byte))
-            {
-                return vector;
-            }
-            else if (typeof(T) == typeof(ushort))
-            {
-                return vector;
-            }
-            else if (typeof(T) == typeof(uint))
-            {
-                return vector;
-            }
-            else if (typeof(T) == typeof(ulong))
-            {
-                return vector;
-            }
-            else
-            {
-                return SoftwareFallback(vector);
-            }
-
-            static Vector256<T> SoftwareFallback(Vector256<T> vector)
-            {
-                Unsafe.SkipInit(out Vector256<T> result);
-
-                for (int index = 0; index < Vector256<T>.Count; index++)
-                {
-                    T value = Scalar<T>.Abs(vector.GetElementUnsafe(index));
-                    result.SetElementUnsafe(index, value);
-                }
-
-                return result;
-            }
+            return Create(
+                Vector128.Abs(vector.GetLower()),
+                Vector128.Abs(vector.GetUpper())
+            );
```

Comment:
This is the software fallback code that will get replaced with the hardware intrinsic if it is available at runtime, correct?

Reply:
Correct.